src/transformers/tokenization_utils_base.py

```diff
@@ -2382,8 +2387,8 @@ def save_pretrained(
         tokenizer_config = copy.deepcopy(self.init_kwargs)
 
         # TODO: Ensure the modified attributes (those are also in the __init__ kwargs) will give identical tokenizers
-        # target_keys = self.init_kwargs.keys()
-        target_keys = ["model_max_length", "clean_up_tokenization_spaces", "additional_special_tokens"]
+        target_keys = list(self.init_kwargs.keys())
```

Comment:
when saving; we should overwrite the init_kwargs with the content of self. Don't know why it was not the case before